### PR TITLE
[CRM-26] Pagination component and story

### PR DESCRIPTION
**Pull request by magikabdul**

### CODE REVIEW

File: frontend/src/components/molecules/Pagination/Pagination.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import KeyboardArrowLeftIcon from '@material-ui/icons/KeyboardArrowLeft';
+import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
+
```
[the remark this answers is not in the document]
Comment (Fellway) [reply frontend/src/components/molecules/Pagination/Pagination.js:6]: Add shadows on hover for numbers

[the remark this answers is not in the document]
Comment (Fellway) [reply frontend/src/components/molecules/Pagination/Pagination.js:6]: Add mouse: pointer on hover for nubmers and arrows

[the remark this answers is not in the document]
Comment (Fellway) [reply frontend/src/components/molecules/Pagination/Pagination.js:6]: If you click on number it should change active element for clicked number

File: frontend/src/components/molecules/Pagination/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      </Number>
+    )}
+
+    {count > 1 && (
+      <Number active={page === 2}>
+        {second(count, page)}
+      </Number>
+    )}
+
+    {count > 2 && (
```
Comment (Fellway) [inline frontend/src/components/molecules/Pagination/Pagination.js:101]: I don't like this logic for generating pages. It should be done in more elegant way.  Code is very unclear

Comment (Fellway) [reply frontend/src/components/molecules/Pagination/Pagination.js:101]: What if you pass only 3 or 2 pages?

File: frontend/src/components/molecules/Pagination/Pagination.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  if (page <= 3) {
+    return '3';
+  }
+  if (page > count - 2) {
+    return count - 2;
+  }
+  return page;
+};
+
+const fourth = (count, page) => {
```
[the remark this answers is not in the document]
Comment (magikabdul) [reply frontend/src/components/molecules/Pagination/Pagination.js:72]: method rewritten

File: frontend/src/components/molecules/Pagination/Pagination.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (magikabdul) [reply frontend/src/components/molecules/Pagination/Pagination.js:6]: done

File: frontend/src/components/molecules/Pagination/Pagination.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+  font-family: 'Roboto', sans-serif;
+  font-weight: bold;
+  font-size: 18px; //TODO
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  background-color: ${({ active }) => (active ? 'blue' : 'transparent')};
```
[the remark this answers is not in the document]
Comment (magikabdul) [reply frontend/src/components/molecules/Pagination/Pagination.js:34]: done

File: frontend/src/components/molecules/Pagination/Pagination.js (hunk 2 above)
Comment (magikabdul) [reply frontend/src/components/molecules/Pagination/Pagination.js:101]: code was rewritten, tested cases with a number of pages 1, 2, 3, 4, 5, and larger

File: frontend/src/components/molecules/Pagination/Pagination.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  color: ${({ active }) => (active ? 'white' : 'inherit')};
```
[the remark this answers is not in the document]
Comment (magikabdul) [reply frontend/src/components/molecules/Pagination/Pagination.js:35]: done

File: frontend/src/components/molecules/Pagination/Pagination.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  justify-content: center;
+  align-items: center;
+`;
+
+const Number = styled.div`
+  width: 32px;
+  height: 32px;
+
+  margin: 8px;
+
```
[the remark this answers is not in the document]
Comment (magikabdul) [reply frontend/src/components/molecules/Pagination/Pagination.js:23]: added

**Review by Fellway — APPROVED** [review]

**Pull request merged**